### PR TITLE
Format titles to break words

### PR DESCRIPTION
**Pull request by hellosnow**
Fix title to break words for #234 #513

### CODE REVIEW

File: src/docfx.website.themes/default/styles/docfx.js [hunk 1]
```diff
@@ -8,6 +8,11 @@ $(function () {
   var hide = 'hide';
   var util = new utility();
 
+  // supported headers are h1, h2, h3, and h4
+  // The topest header is ignored
+  var articleSelector = ".article article";
```
Comment (vicancy) [inline src/docfx.website.themes/default/styles/docfx.js]: >".article article" [](start = 24, length = 18)

article is enough?

Comment (hellosnow) [inline src/docfx.website.themes/default/styles/docfx.js]: want to select "article" element under "article" class?

Comment (vicancy) [inline src/docfx.website.themes/default/styles/docfx.js]: using semantic tag article is enough?

---
In reply to: [118413863](https://github.com/dotnet/docfx/pull/1695#discussion_r118413863) [](ancestors = 118413863)

File: src/docfx.website.themes/default/styles/docfx.js [hunk 2]
```diff
@@ -32,6 +34,14 @@ $(function () {
     renderAffix();
   }
 
+  // Format titles to break words
+  function formatTitles() {
+    var headers = $("article h1, article h2, article h3, article h4");
+    for (var j = 0; j < headers.length; j++) {
+      headers[j].textContent = util.breakText(headers[j].textContent);
```
Comment (vicancy) [inline src/docfx.website.themes/default/styles/docfx.js]: >textContent [](start = 17, length = 11)

use jquery's text() for cross-browser support?

Comment (hellosnow) [inline src/docfx.website.themes/default/styles/docfx.js]: yes that's better

**Comment by vicancy** [comment]
:shipit:

File: src/docfx.website.themes/default/styles/docfx.js [hunk 3]
```diff
@@ -32,6 +34,16 @@ $(function () {
     renderAffix();
   }
 
+  // Format titles to break words
+  function formatTitles() {
+    var headers = $("article h1, article h2, article h3, article h4");
```
Comment (vwxyzh) [inline src/docfx.website.themes/default/styles/docfx.js]: consider: "a.xref, span.xref"

Comment (hellosnow) [inline src/docfx.website.themes/default/styles/docfx.js]: add a new class break-text and apply it for xref class

---
In reply to: [118621480](https://github.com/dotnet/docfx/pull/1695#discussion_r118621480) [](ancestors = 118621480)

File: src/docfx.website.themes/default/partials/rest.tmpl.partial [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {{!Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.}}
 
-<h1 id="{{htmlId}}" data-uid="{{uid}}" class="text-capitalize">{{name}}</h1>
+<h1 id="{{htmlId}}" data-uid="{{uid}}" class="text-capitalize break-text">{{name}}</h1>
```
Comment (vicancy) [inline src/docfx.website.themes/default/partials/rest.tmpl.partial]: follow the same naming convention as *text-capitalize*, to *text-break*?

File: src/docfx.website.themes/default/styles/docfx.css [hunk 5]
```diff
@@ -37,9 +37,9 @@ h4:before {
     margin-top: -100px;
 }
 
-h1, h2, h3, h4, h5, h6, span.xref {
+h1, h2, h3, h4, h5, h6, span.xref, .text-break {
```
Comment (superyyrrzz) [inline src/docfx.website.themes/default/styles/docfx.css]: >span.xref [](start = 24, length = 9)

remove

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**